File: event-sdk-generator/packages/runtime-typescript/src/adapters/memory.ts
import type { TransportAdapter, EventEnvelope, PublishResult } from "../types.js";

export type RecordedEvent = {
  envelope: EventEnvelope;
  publishedAt: Date;
};

export class InMemoryTransportAdapter implements TransportAdapter {
  readonly published: RecordedEvent[] = [];

  async publish(envelope: EventEnvelope): Promise<PublishResult> {
    const messageId = `mem-${Date.now()}-${Math.random().toString(36).slice(2)}`;
    this.published.push({ envelope, publishedAt: new Date() });
    return { messageId };
  }

  clear(): void {
    this.published.length = 0;
  }
}

File: event-sdk-generator/packages/runtime-python/src/eventgen_runtime/adapters/memory.py
from __future__ import annotations
from dataclasses import dataclass, field
from datetime import datetime, timezone
import uuid
from eventgen_runtime.types import EventEnvelope, PublishResult, TransportAdapter


@dataclass
class RecordedEvent:
    envelope: EventEnvelope
    published_at: datetime


class InMemoryTransportAdapter:
    def __init__(self) -> None:
        self.published: list[RecordedEvent] = []

    async def publish(self, envelope: EventEnvelope) -> PublishResult:
        self.published.append(RecordedEvent(envelope=envelope, published_at=datetime.now(timezone.utc)))
        return PublishResult(message_id=str(uuid.uuid4()))

    def clear(self) -> None:
        self.published.clear()
